src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -2204,11 +2207,15 @@ internal override void IntersectWithEnumerable(IEnumerable<T> other)
                     if (Contains(item))
                     {
                         toSave.Add(item);
-                        Remove(item);
                     }
                 }
-                Clear();
-                AddAllElements(toSave);
+
+                if (toSave.Count < Count)
+                {
+                    Clear();
+                    AddAllElements(toSave);
+                }
+
```

Comment:
Only slightly related to your change, anyone know why TreeSubSet overrides InterSectWithEnumerable?  Other than the debug-only assert at the end, it looks identical in implementation to the base.  What am I missing at brief glance?  It'd be nice to only make this change in one rather than two places.
cc: @ianhays